server/server.go

```diff
@@ -105,19 +105,27 @@ func (srv *Server) Start() {
 }
 
 func (srv *Server) init(cfg *Config) {
-	dp := fmt.Sprintf("%s/db", cfg.DataDir)
-	err := storage.EnsureDir(dp, 0700)
-	if err != nil {
-		log.Fatalf("could not init sqlite path: %s", err)
+	if cfg.DB == nil {
+		dp := fmt.Sprintf("%s/db", cfg.DataDir)
```

Comment:
Careful, we need to use `filepath.Join` here, otherwise we're generating invalid path names on non-Unix platforms.